### PR TITLE
Allow the EnumSwitch API to take a default

### PR DESCRIPTION
**Pull request by ashfall**
This also provides a default value to `_constructs.Extension`.

Addresses the issues identified in https://github.com/pyca/tls/pull/105#issuecomment-261779992.

### CODE REVIEW

File: tls/_common/_constructs.py [hunk 1]
```diff
@@ -193,13 +193,21 @@ def EnumSwitch(type_field, type_enum, value_field, value_choices):  # noqa
         match any members without an explicit mapping.
     :type value_choices: :py:class:`dict`
 
+    :param default: A default field to use when no explicit match is found
+        for the key in the provided mapping. This follows
+        :py:func:`construct.core.Switch`'s API, so if not supplied, an
+        exception will be raised when the key is not found. Pass can be used
```
Comment (markrwilliams) [inline tls/_common/_constructs.py]:
```
:py:class:`construct.Pass`
```

File: tls/_common/_constructs.py [hunk 2]
```diff
@@ -149,7 +149,7 @@ def EnumClass(type_field, type_enum):  # noqa
     return construct.SymmetricMapping(type_field, mapping)
 
 
-def EnumSwitch(type_field, type_enum, value_field, value_choices):  # noqa
+def EnumSwitch(type_field, type_enum, value_field, value_choices, default=construct.Switch.NoDefault):  # noqa
```
Comment (markrwilliams) [inline tls/_common/_constructs.py]: I think I added `# noqa` so that we could follow construct's NamingConventionForFunctions without upsetting pylint.  Two questions:  

1) Maybe it's better to rename these _following_the_standard_convention?  (This would be in a separate PR)
2) Do you think it makes sense to wrap the signature at 79 characters to stay consistent with the rest of the project?

Comment (ashfall) [inline tls/_common/_constructs.py]: Thanks a lot for the review! 

Yes re: the reason we have `# noqa` here.

1. Probably? I don't have any strong opinions about this, I might not have enough bandwidth to do a refactor/rename of these things _and_ work on the extensions at the moment, and if I had to pick one, I would pick the extensions work. But, feel free to file a ticket / send a PR, happy to review! :)

2. Yeah, makes sense. Updating.

File: tls/_common/_constructs.py (hunk 1 above)
Comment (ashfall) [inline tls/_common/_constructs.py]: Thanks, good catch!

**Comment by markrwilliams** [comment]
@ashfall Thanks so much for doing this!  Merging so #105 can move forward.

File: tls/_common/_constructs.py (hunk 2 above)
Comment (ashfall) [inline tls/_common/_constructs.py]: Yay!! Thanks so much! :)
